dhall/ghc-src/Dhall/Import/HTTP.hs

```diff
@@ -166,6 +169,21 @@ newManager = do
         Just manager ->
             return manager
 
+getSiteHeaders :: StateT Status IO SiteHeaders
+getSiteHeaders = do
+    Status { _siteHeaders = oldSiteHeaders, _stack, ..} <- State.get
+
+    case oldSiteHeaders of
+        Nothing -> do
+            siteHeaders <- liftIO (_loadSiteHeaders _stack)
```

Comment:
I think you want `_loadSiteHeaders` to reuse the ambient `StateT Status` here.  Then you don't need to pass the `_stack` to it explicitly.

Also, then you wouldn't need to extend the `Status` to add a new `_loadSiteHeaders` field

Reply:
I think I still need the _loadSiteHeaders field to the status in order to support creating a status based on `envOnlyUserHeaders` (in the tests), `defaultUserHeaders` normally, and `noopUserHeaders` (for remoteStatus).

Is there another way to do that without keeping it in the status? And when you say "reuse the ambient StateT Status", do you mean to make `_loadSiteHeaders` be type `StateT Status IO SiteHeaders` (My eyes still fog over when I read monad transformer types 😅 )?